// react-mailto-link/lib/esm/index.js
import { jsx as _jsx } from "react/jsx-runtime";
export const headersToRequest = (headers) => {
    return Object.entries(headers)
        .map(([key, value]) => {
        return `${key}=${encodeURIComponent(value)}`;
    })
        .join("&");
};
export const createMailtoLink = ({ email, headers }) => {
    if (!headers) {
        return `mailto:${email}`;
    }
    return `mailto:${email}?${headersToRequest(headers)}`;
};
const Mailto = ({ children, className, headers, email, obfuscated, style, }) => {
    function handleClick(e) {
        e.preventDefault();
        window.open(createMailtoLink({ email, headers }));
    }
    const Obfuscate = (props) => (_jsx("span", Object.assign({ style: {
            unicodeBidi: "bidi-override",
            direction: "rtl",
            WebkitTransform: "rotateY(180deg)",
            msTransform: "rotateY(180deg)",
            OTransform: "rotateY(180deg)",
        } }, { children: props.email.split("").reverse().join("") }), void 0));
    return obfuscated ? (_jsx("a", Object.assign({ onClick: handleClick, href: "mailto:obfuscated", className: className, style: style }, { children: children || _jsx(Obfuscate, { email: email }, void 0) }), void 0)) : (_jsx("a", Object.assign({ href: `mailto:${email}`, className: className, style: style }, { children: children || email }), void 0));
};
export default Mailto;
