// react-mailto-link/lib/cjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMailtoLink = exports.headersToRequest = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const headersToRequest = (headers) => {
    return Object.entries(headers)
        .map(([key, value]) => {
        return `${key}=${encodeURIComponent(value)}`;
    })
        .join("&");
};
exports.headersToRequest = headersToRequest;
const createMailtoLink = ({ email, headers }) => {
    if (!headers) {
        return `mailto:${email}`;
    }
    return `mailto:${email}?${exports.headersToRequest(headers)}`;
};
exports.createMailtoLink = createMailtoLink;
const Mailto = ({ children, className, headers, email, obfuscated, style, }) => {
    function handleClick(e) {
        e.preventDefault();
        window.open(exports.createMailtoLink({ email, headers }));
    }
    const Obfuscate = (props) => (jsx_runtime_1.jsx("span", Object.assign({ style: {
            unicodeBidi: "bidi-override",
            direction: "rtl",
            WebkitTransform: "rotateY(180deg)",
            msTransform: "rotateY(180deg)",
            OTransform: "rotateY(180deg)",
        } }, { children: props.email.split("").reverse().join("") }), void 0));
    return obfuscated ? (jsx_runtime_1.jsx("a", Object.assign({ onClick: handleClick, href: "mailto:obfuscated", className: className, style: style }, { children: children || jsx_runtime_1.jsx(Obfuscate, { email: email }, void 0) }), void 0)) : (jsx_runtime_1.jsx("a", Object.assign({ href: `mailto:${email}`, className: className, style: style }, { children: children || email }), void 0));
};
exports.default = Mailto;
